nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/resources/docs/org.apache.nifi.processors.script.ScriptedTransformRecord/additionalDetails.html

```diff
@@ -0,0 +1,412 @@
+    That Record is then written using the configured Record Writer. If the script returns a <code>null</code> value, the Record will not be written. If the script returns an object that is not
+    a Record, the incoming FlowFile will be routed to the <code>failure</code> relationship.
+</p>
+
+<p>
+    This processor maintains two Counters: "Records Transformed" indicating the number of Records that were passed to the script and for which the script returned a Record, and "Records Dropped"
+    indicating the number of Records that were passed to the script and for which the script returned a value of <code>null</code>.
+</p>
+
+
+
+<h3>Variable Bindings</h3>
+
+<p>
+    While the script provided to this Processor does not need to provide boilerplate code or implement any classes/interfaces, it does need some way to access the Records and other information
+    that it needs in order to perform its task. This is accomplished by using Variable Bindings. Each time that the script is invoked, each of the following variables will be made
+    available to the script:
+</p>
+
+<table>
+    <tr>
+        <th>Variable Name</th>
+        <th>Description</th>
+        <th>Variable Class</th>
+    </tr>
+    <tr>
+        <td>record</td>
+        <td>The Record that is to be transformed.</td>
+        <td><a href="https://javadoc.io/static/org.apache.nifi/nifi-record/1.11.4/org/apache/nifi/serialization/record/Record.html">Record</a></td>
+    </tr>
+    <tr>
+        <td>recordIndex</td>
+        <td>The zero-based index of the Record in the FlowFile.</td>
+        <td>Long (64-bit signed integer)</td>
+    </tr>
+    <tr>
+        <td>log</td>
+        <td>The Processor's Logger. Anything that is logged to this logger will be written to the logs as if the Processor itself had logged it. Additionally, a bulletin will be created for any
+            log message written to this logger (though by default, the Processor will hide any bulletins with a level below WARN).</td>
+        <td><a href="https://www.javadoc.io/doc/org.apache.nifi/nifi-api/latest/org/apache/nifi/logging/ComponentLog.html">ComponentLog</a></td>
+    </tr>
+    <tr>
+        <td>attributes</td>
+        <td>Map of key/value pairs that are the Attributes of the FlowFile. Both the keys and the values of this Map are of type String. This Map is immutable.
+            Any attempt to modify it will result in an UnsupportedOperationException being thrown.</td>
+        <td>java.util.Map</td>
+    </tr>
+</table>
+
+
+<a name="ReturnValue"></a>
+<h3>Return Value</h3>
+
+<p>
+    Each time that the script is invoked, it is expected to return a
+    <a href="https://javadoc.io/static/org.apache.nifi/nifi-record/1.11.4/org/apache/nifi/serialization/record/Record.html">Record</a> object.
+    That Record is then written using the configured Record Writer. If the script returns a <code>null</code> value, the Record will not be written. If the script returns an object that is not
```

Comment:
Should add a sentence here describing how a Collection of Records can be returned and all will be written out.